### PR TITLE
343-340-348- step 3 component and interactive footer buttons

### PR DESCRIPTION
**Pull request by anaislet-c2c**
### JIRA issue

https://jira.camptocamp.com/browse/GSREN3D-343-340-348

### Description

step 3 component added and footer buttons fixed

### Screenshots

image

### CODE REVIEW

File: src/components/simulation/BoxStep3.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<script lang="ts" setup>
```
Comment (toni-garcia-camptocamp) [inline src/components/simulation/BoxStep3.vue:1]: Component BoxStep2 and BoxStep3 are almost the same.
Can you create a generic component named for exemple 'BoxStep' and use it in BoxStep2 and BoxStep3 please
This avoids duplicating the code

File: src/components/simulation/ExplanationSelectionObstacles.vue [hunk 2]
```diff
@@ -2,10 +2,21 @@
 import step2 from '@/assets/illustrations/Steps.jpg'
```
Comment (toni-garcia-camptocamp) [inline src/components/simulation/ExplanationSelectionObstacles.vue:2]: The component ExplanationSelectionObstacles.vue is only for this part:
image
To manage the substep of step 2 can you create an another component please

**Review by toni-garcia-camptocamp — COMMENTED** [review]
Nice work, just somechange to factorize code

File: src/components/simulation/FooterButtons.vue [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
 const router = useRouter()
 const simulationStore = useSimulationStore()
-
 function clickButtonCancel() {
   router.push('/roof-selection')
 }
-
 function clickButtonPrevious() {
-  simulationStore.goToPreviousStep()
+  if (simulationStore.currentStep == 3) {
```
Comment (ismailsunni) [inline src/components/simulation/FooterButtons.vue:10]: I think we should put this navigation in the simulation store, and the button should only call `goToPreviousStep()` or `gotToNextStep()`

File: src/stores/simulations.ts [hunk 4]
```diff
@@ -16,18 +26,53 @@ export const useSimulationStore = defineStore('simulation', () => {
     }
   }
 
+  function setCurrentSubStep(subStep: number) {
+    if (subStep < 0) {
+      currentSubStep.value = 0
+    } else if (subStep > 2) {
```
Comment (toni-garcia-camptocamp) [inline src/stores/simulations.ts:32]: Here of the next is step 3 and substep 3 , the substep is set to 2. It's not good

File: src/stores/simulations.ts [hunk 5]
```diff
@@ -16,18 +26,53 @@ export const useSimulationStore = defineStore('simulation', () => {
...
+      currentSubStep.value = 2
+    } else {
+      currentSubStep.value = subStep
+    }
+  }
+
   function goToPreviousStep() {
```
Comment (toni-garcia-camptocamp) [inline src/stores/simulations.ts:39]: I think you can replace this function by something like this :

`function indexOfCurrentStepAndSubStep() {
    return availableSteps.findIndex((availableStep) => {
      return availableStep.step == currentStep.value && availableStep.subStep == currentSubStep.value
    })
  }

  function goToPreviousStep() {
    const currentIndex = indexOfCurrentStepAndSubStep()
    if(currentIndex-1 < 0) {
      return
    }
    setCurrentStep(availableSteps[currentIndex-1].step)
    setCurrentSubStep(availableSteps[currentIndex-1].subStep)
  }`

File: src/stores/simulations.ts (hunk 4 above)
Comment (toni-garcia-camptocamp) [reply src/stores/simulations.ts:32]: You can just make this function on one line : currentSubstep.value = subStep
Same for function setCurrentStep

**Pull request merged**